### DISCUSSION TITLE
Is QuML a standard for content (apart from assessment), and whether it is based on some globally accepted standard. Also which all telemetry events are possible not possible with without SCORM/TINCAN/QuML ?

### DISCUSSION THREAD

**Discussion by sujitrs** [sunbird-specs/QuML#6] [Q&A]
Wants to know if QuML is also a standard for content (apart from assessment), and whether it is based on some globally accepted standard. Also which all telemetry events are possible not possible with without SCORM/TINCAN/QuML ?

**Comment by pramodkvarma** [comment]
Currently QuML is only meant to capture questions in a markup structure to achieve interoperability across learning management and assessment systems. 

But the way QuML is constructed, any kind of questions (practice, assessment, quizzes, polls, surveys, tests, etc) can be captured. 

The generalized and flexible nature of QuML is powerful so that solutions and UX can be separated out from representation and storage. As I said earlier interoperability of QuML also allows solutions/apps to be separate from the platform that provides questions.

**Comment by pramodkvarma** [comment]
SCROM is a content storage and exchange mechanism. Sunbird can easily support a SCROM adapter to bring in/out content from other systems. There is an open source implementation of SCROM adapter already available for Sunbird.

Tincan and telemetry are comparable. I don't think we have done extensive analysis comparing them. I think [Sunbird telemetry specs](https://telemetry.sunbird.org/) support lot more capability beyond Tincan which only focuses on learning related telemetry. Sunbird is not merely designed for traditional LMS purposes and hence Sunbird telemetry accommodates many types of interaction events inclusive system level telemetry.

Maybe you can do it and submit here. Community may find ot useful.

**Reply by sujitrs** [reply]
So here is the approach that needs to be taken to have straight answer to the question that is raised:

Need to study what SCORM, tincan and Sunbird Telemetry specs has to offer and then conclude on the question raised.

A quick look at SCORM gives below information but again without in-depth understanding there is no meaning to this data as well...

As per [SCORM article](https://scorm.com/scorm-explained/business-of-scorm/scorm-versions/), even SCORM's latest version doesn't support cross domain working, but at the same time it is widely used, it makes creating sequenced content much simpler, supports [run-time](https://scorm.com/scorm-explained/technical-scorm/run-time/),  [packaging](https://scorm.com/scorm-explained/technical-scorm/content-packaging/), [metadata](https://scorm.com/scorm-explained/technical-scorm/content-packaging/metadata-structure/),  [sequencing](https://scorm.com/scorm-explained/technical-scorm/sequencing/sequencing-definition-model/).

A detailed study is expected to conclude. Hence leaving it here as unanswered.

**Comment by parthlawate** [comment]
Syncing discussions elsewhere on Github for visibility to the larger community. 

**Background Context** 

IMS Question and Test Interoperability (IMS QTI) is the best-known standard for tests. However it has some issues from a practical implementation standpoint which make interoperability harder.. QuML is derived from QTI and addresses these issues.

More about QuML can be read here https://quml.sunbird.org/

QuML uses Sunbird telemetry for tracking..more on that here  https://quml.sunbird.org/v1/telemetry

A well written SCORM can also provide some question level tracking in terms of write and wrong answers but due to lack of standardisation it's not very interoperable. 

Also from an actual adoption perspective SCORM having got wider acceptance will have more legacy content and that we might plan to support. In comparison I've seen very less volume of  QTI compatible content.

**State of current SCORM adapter for Sunbird**
This was made for a very specific usecase where content exported from an Authoring tool as HTML5 object with an entry file named as index_lms.html. It currently is able to store progress & allow resume using the ContentState Update API to allow resume. This needs to be extended to allow using SCORM objects from any standard authoring tool & allow for wider range of tracking. 

**Broken down Question** 
A. is QuML is also a standard for content (apart from assessment) ?  
**It mainly handles assessments as of now**

B. is it  based on some globally accepted standard. 
**Yes IMS QTI**

C. Also which all telemetry events are possible /not possible with /without SCORM/TINCAN/QuML ? 

I've started a comparison between SCORM, Tincan & Sunbird Telemetry -- those are comparable. 
If we want to compare QuML it will need to be compared with QTI I believe.  

The way I see it is that Sunbird Telemetry can be looked at as a 'Superset' and if we use SCORM /TinCan content we can write adaptors which can sync the 'standard' tracking from each into Sunbird Telemetry.  Bulk of content available in the corporate space is SCORM so prioritising this adaptor would be a good idea.  We should look into 'cmi5' which is kind of a bridge between xAPI (which is vast overall) & seems like a good base to decide what tracking from SCORM should be synced to Sunbird Telemetry. 

I'll post another update once i have a more detailed comparison ready.
